src/raftstore/store/store.rs

```diff
@@ -2269,15 +2269,19 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                 );
                 return;
             }
-            Some(p) => (&mut p.consistency_state, &p.peer),
+            Some(peer) => peer,
         };
 
-        if !verify_and_store_hash(region_id, state, index, hash) {
+        if !verify_and_store_hash(region_id, &mut peer.consistency_state, index, hash) {
             return;
         }
 
+        if !peer.is_leader() {
```

Comment:
What if the peer is not leader now but become leader before receiving verify hash request?

Reply:
any update @coder0718